src/dioptra/restapi/v1/models/service.py

```diff
@@ -550,14 +571,23 @@ def create(
 
         ml_model.resource.children.append(new_version.resource)
         db.session.add(new_version)
+        db.session.flush()
 
-        if commit:
-            db.session.commit()
-            log.debug(
-                "Model registration successful",
-                model_id=ml_model.resource_id,
-                name=ml_model.name,
+        mlflow_model_name = f"resource_{ml_model.resource_id:09d}"
+        try:
+            self._mlflow_client.create_model_version(
+                mlflow_model_name, source=artifact.uri
             )
+        except MlflowException as e:
+            raise MLFlowError(e.message) from e
+
+        db.session.commit()
```

Comment:
Is it possible to delete a model version after you've created it in Mlflow? If so, then we need to wrap this in an try/except to allow rollbacks? If not, then I'm not sure how we handle the inconsistent state.